samcli/lib/providers/sam_api_provider.py

```diff
@@ -207,15 +215,14 @@ def _convert_event_route(stack_path: str, lambda_logical_id, event_properties, e
         :param event_type: The event type, 'Api' or 'HttpApi', see samcli/local/apigw/local_apigw_service.py:35
         :return tuple: tuple of route resource name and route
         """
-        path = event_properties.get(SamApiProvider._EVENT_PATH)
-        method = event_properties.get(SamApiProvider._EVENT_METHOD)
+        path = cast(str, event_properties.get(SamApiProvider._EVENT_PATH))
+        method = cast(str, event_properties.get(SamApiProvider._EVENT_METHOD))
 
         # An RESTAPI (HTTPAPI) Event, can have RestApiId (ApiId) property which designates the resource that owns this
         # API. If omitted, the API is owned by Implicit API resource. This could either be a direct resource logical ID
         # or a "Ref" of the logicalID
 
-        api_resource_id = None
```

Comment:
why did we remove this?

Reply:
Declaring `api_resource_id` as None is not necessary since it is overridden in the next few lines. Making it None confused mypy making it thinking it is an Optional object